library/std/src/path.rs

```diff
@@ -979,6 +979,19 @@ impl FusedIterator for Components<'_> {}
 impl<'a> cmp::PartialEq for Components<'a> {
     #[inline]
     fn eq(&self, other: &Components<'a>) -> bool {
+        // fast path for exact matches, e.g. for hashmap lookups
+        if self.path.len() == other.path.len()
+            && self.front == other.front
+            && self.back == State::Body
+            && other.back == State::Body
+        {
+            // this could bail out earlier if there were a reverse memcmp/bcmp comparing back to front
```

Comment:
Can we phrase this as a "future improvement" or something? Currently, it's not bailing out early, right? This comment makes that a little unclear.